src/util/string.h

```diff
@@ -159,6 +162,37 @@ inline bool str_equal(const std::basic_string<T> &s1,
 	return true;
 }
 
+// For some reason an std::string will not implicitly get converted
+// to an std::basic_string_view<char> in the template case above, so we need
+// these three wrappers. It works if you take out the template parameters.
+// see also <https://stackoverflow.com/questions/68380141/>
```

Comment:
I'm not sure what the cleanest way to go about this is, but macros definitely seem appropriate, especially since we also want this for starts with.

We have the string view based template function, which we want to define other functions in terms of. So how about something like (untested):

```c++
#define MAKE_VARIANT(func, T1, T2) \
  template <typename T> \
  inline bool func(T1 s1, T2 s2, bool case_insensitive = false) \
  { \
    return func(std::basic_string_view<T>(s1), std::basic_string_view<T>(s2), case_insensitive); \
  }
```

Then

```cpp
#define MAKE_VARIANTS(func) \
  MAKE_VARIANT(func, std::basic_string_view<T>, const std::basic_string<T>&); \
  MAKE_VARIANT(func, const std::basic_string<T>&, std::basic_string_view<T>); \
  MAKE_VARIANT(func, const std::basic_string<T>&, const std::basic_string<T>&);
MAKE_VARIANTS(str_equal)
MAKE_VARIANTS(str_starts_with)
MAKE_VARIANTS(str_ends_with)
```

(+ `#undef` at the end since this is a header).

The `std::basic_string_view` copy constructors should be optimized out if `T1` / `T2` is already a string view.